test/phoenix/router/helpers_test.exs

```diff
@@ -221,6 +244,51 @@ defmodule Phoenix.Router.HelpersTest do
     assert Helpers.user_comment_path(__MODULE__, :show, 123, 2) == "/users/123/comments/2"
     assert Helpers.user_comment_path(__MODULE__, :new, 88, []) == "/users/88/comments/new"
     assert Helpers.user_comment_path(__MODULE__, :new, 88) == "/users/88/comments/new"
+
+    error_message = (fn helper ->
+    """
+    No route helper clause for Phoenix.Router.HelpersTest.Router.Helpers.#{helper} defined for action :skip.
+    The following #{helper} actions are defined under your router:
+
+     * :create
+     * :delete
+     * :edit
+     * :index
+     * :new
+     * :show
+     * :update
+    """ |> String.strip
+    end)
+
+    assert_raise ArgumentError, error_message.("user_comment_path/3"), fn ->
+      Helpers.user_comment_path(__MODULE__, :skip, 123)
+    end
+
+    assert_raise ArgumentError, error_message.("user_comment_file_path/4"), fn ->
+      Helpers.user_comment_file_path(__MODULE__, :skip, 123, 456)
+    end
+
+    assert_raise ArgumentError, error_message.("user_comment_file_path/5"), fn ->
+      Helpers.user_comment_file_path(__MODULE__, :skip, 123, 456, foo: "bar")
+    end
+
+    arity_error_message =
+    """
+    No route helper clause for Phoenix.Router.HelpersTest.Router.Helpers.user_comment_path/3 defined for action :show with an arity of 3. Please check that the function, arity and action are correct.
+   The following user_comment_path/3 actions are defined under your router:
+
+     * :create
+     * :delete
+     * :edit
+     * :index
+     * :new
+     * :show
+     * :update
+    """ |> String.strip
```

Comment:
Since we have all the routes, we could even print the `ConsoleFormatter` version of the routes for only this particular helper, that way the user gets something like:

```
No route helper clause for MyApp.Router.Helpers.user_post_path/3 defined
for action :show with an arity of 3.
Please ensure you are passing all dynamic segments of the path as arguments.
The following user_post_path/3 actions are defined under your router:

   user_post_path  GET     /users/:user_id/post/:id       MyApp.PostController :index
   user_post_path  GET     /users/:user_id/posts/new      MyApp.PostController :new
   user_post_path  GET     /users/:user_id/post/:id       MyApp.PostController :show
   user_post_path  POST    /users/:user_id/posts          MyApp.PostController :create
```